pkg/gencred/README.md

```diff
@@ -0,0 +1,100 @@
+# Gencred
+
+## Description
+
+`gencred` is a simple tool used to generate cluster auth credentials (w/ [**cluster-admin**](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#user-facing-roles) permissions) for authenticating to a Kubernetes cluster.
+> **NOTE:** since `gencred` creates credentials with `cluster-admin` level access, the kube context used **must** also be bound to the `cluster-admin` role.
+
+## Usage
+
+### Generate a service account token for a cluster. 
+✅ **PREFERRED** method.
+
+```go
+// Import gencred
+import "k8s.io/test-infra/pkg/gencred"
+
+//...
+
+// Create a Kubernetes clientset for interacting with the cluster.
+// In this case we are simply using the `current-context` defined in our local `~/.kube/config`.
+homedir, _ := os.UserHomeDir()
+kubeconfig := filepath.Join(homedir, ".kube", "config")
+config, _ := clientcmd.BuildConfigFromFlags("", kubeconfig)
+clientset, _ := kubernetes.NewForConfig(config)
+
+// Generate a service account token, as well as return the certificate authority that issued the token.
+token, caPEM, err := gencred.CreateClusterServiceAccountCredentials(clientset)
+```  
+
+`token` will contain the **service account access token** and `caPEM` will contain the **server certificate authority**.
+
+```go
+import "encoding/base64"
+
+//...
+
+// Cast the `token` to a string to use in a kubeconfig.
+accessToken := string(token)
+// Base64 encode the `caPEM` to use in a kubeconfig.
+ca := base64.StdEncoding.EncodeToString(caPEM)
+
+fmt.Println("token:", accessToken)
+fmt.Println("ca:", ca)
+```
+
+```text
+token: eyJhbGciOiJSUzI1NiIsImtpZCI6IiJ9.eyJpc3Mit...
+ca: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSURER...
+```
+
+### Generate a client key and certificate for a cluster.
```

Comment:
Maybe we just want to delete this?

Reply:
I propose we hold on to it *at least* until all the Prow instances we manage have switched over to using a `kubeconfig` file rather `build-cluster` format. This will be **needed** for the later. Additionally, this is the only tool that I am aware of that can do this with a *single* function, so it may be generally useful.